architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,206 @@
+# Radius Applications RP Component Threat Model
+
+- **Author**: @nithyatsu
+
+## Overview
+
+This document provides a threat model for the Radius Applications RP component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+The Applications RP component is responsible for managing applications and their resources. It communicates with UCP, Deployment Engine and Controller components to achieve this. 
+
+## Terms and Definitions
+
+| Term                  | Definition                                                                                                                                                                                  |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| mTLS                  | Mutual Transport Layer Security (mTLS) allows two parties to authenticate each other during the initial connection of an SSL/TLS handshake.                                                 |
+| UCP                 | Universal Control Plane for Radius                                                                                                                                                   |
+## System Description
+
+Applications RP is a Radius service that acts as resource provider for application and its resources. The resources can be core resources like application or environment or container. They can also be a dapr resource, message queue or datastore. Applications RP lives in `radius-system` namespace in a kubernetes cluster. It is a client of Controller and UCP. It also receives requests from UCP for managing the above mentioned resources. 
```

Comment:
I think we're missing a few points. Here's the big headlines I'd want to see covered. These are all **relevant** to the security discussion.

- Applications RP is a resource provider. Resource providers communicate over HTTP. 
- Applications RP needs a database, message queue, and secret store to function. These are all configurable components with multiple implementations.
- Users and clients do not interact with Applications RP directly. They communicate with UCP instead. 
- Applications RP manages resources in Kubernetes on the user's behalf. This may launch application code in the same Kubernetes cluster as Radius, or a different cluster.
- Applications RP can invoke *recipes* that execute Bicep or Terraform code.
- Applications RP has access to cloud credentials, and can manage cloud resources on the user's behalf.